plugins/spamassassin.js

```diff
@@ -44,6 +44,9 @@ exports.hook_data_post = function (next, connection) {
     const plugin = this;
     if (plugin.msg_too_big(connection)) return next();
 
+    const header_name = plugin.cfg.main.spamc_auth_header || 'X-Haraka-Relay';
+    connection.transaction.remove_header(header_name); // just to be safe
+
```

Comment:
Instead of having duplicate code for the auth_header default name, include it instead in the config loader so the value is always set. There's already a block in `load_spamassassin_ini()` called `defaults` to insert it into:

```js
$ git diff plugins/spamassassin.js 
diff --git a/plugins/spamassassin.js b/plugins/spamassassin.js
index 87d2704a..abcde956 100644
--- a/plugins/spamassassin.js
+++ b/plugins/spamassassin.js
@@ -24,6 +24,7 @@ exports.load_spamassassin_ini = function () {
         max_size:     500000,
         old_headers_action: "rename",
         subject_prefix: "*** SPAM ***",
+        spamc_auth_header: 'X-Haraka-Relay',
     };
 
     for (const key in defaults) {
```js